common/client/node.go

```diff
@@ -43,6 +44,14 @@ type NodeConfig interface {
 	SelectionMode() string
 	SyncThreshold() uint32
 	NodeIsSyncingEnabled() bool
+	FinalizedBlockPollInterval() time.Duration
+}
+
+type ChainConfig interface {
```

Comment:
Can you help me reason through why this configuration belongs to `ChainConfig` vs `NodeConfig` or if `ChainConfig` is living in the correct place (vs in MultiNode)? 

I'd expect configuration here to be node specific and for chain details to live at a higher level in the abstraction hierarchy, or for `node` to store a `map[string]*ChainConfig` to support multiple chain configurations

Reply:
`FinalizedBlockPollInterval` belongs to `NodeConfig` and defines how often a `Node` instance should poll for a new finalized block. It's not a property of a chain; it's a property of the component that performs the health assessment of an RPC.

Node is responsible for the health assessment of a single RPC that works only with one chain. Node does not store `map[string]*ChainConfig` as there is no reason for it to be aware of other chain configurations.